src/ameba/rule/lint/trailing_rescue_exception.cr

```diff
@@ -0,0 +1,51 @@
+module Ameba::Rule::Lint
+  # A rule that prohibits the common misconception about trailing rescue statements,
+  # preventing the use of Paths (exception class names or otherwise) from being
+  # used as the trailing value.
+  #
+  # For example, this is considered invalid:
+  #
+  # ```
+  # value = method.call("param") rescue MyException
+  # ```
+  #
+  # And should instead be written as:
+  #
+  # ```
+  # value = begin
+  #   method.call("param")
+  # rescue MyException
+  #   "default value"
+  # end
+  # ```
```

Comment:
Since this doc is shown when using `explain` command, I would suggest to add some better explanation about misconception here. There is an example, which is good, but maybe we could explain what is actually wrong here?

Reply:
I've updated this, let me know if there's more information / clarity that should be added.